### PR TITLE
Language Server: Suggest completion items for declaration ranges

### PR DESCRIPTION
**Pull request by turbolent**
Blocked on onflow/flow-go#710

## Description

Suggest completion items for the new declaration ranges:

Screen Shot 2021-05-08 at 2 40 12 PM

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by janezpodhostnik — APPROVED** [review]
I don't have a lot of experience in this area, but the code looks good to me.

File: languageserver/server/server.go [hunk 1]
```diff
@@ -872,7 +874,7 @@ var containerCompletionItems = []*protocol.CompletionItem{
 // Completion is called to compute completion items at a given cursor position.
 //
 func (s *Server) Completion(
-	_ protocol.Conn,
+	conn protocol.Conn,
```
Comment (turbolent) [inline languageserver/server/server.go:877]:
```suggestion
	_ protocol.Conn,
```

**Comment by turbolent** [comment]
Blocked by an update of the CLI to the latest Cadence version, so implicitly blocked on the update of flow-go / FVM (onflow/flow-go#710)

**Comment by turbolent** [comment]
@MaxStalker did you accidentally push to this branch? It suddenly have lots of commits, lots of conflicts ... and rebasing and merging both fail

**Pull request merged**